### DISCUSSION TITLE
QUIC packets in transparent mode (commit ff0155b)

### DISCUSSION THREAD

**Discussion by josemariafr-upm** [mitmproxy/mitmproxy#6431] [Q&A]
Hi everyone. I've been trying to make work the transparent mode for QUIC packets using the commit [ff0155b](https://github.com/mitmproxy/mitmproxy/commit/ff0155b1f79453c1dd444f28333fee5baf2ba3af) that @mhils told me but no luck, and I haven't found the reason.
What I'm trying to do is: I'm running a Caddy server with a simple HTML index.html, and I'm doing curl requests to it (with --http3-only option to force the HTTP/3 exchange) to then capture the packets both with Wireshark and MitMProxy (at least the QUIC packets since HTTP/3 is not yet supported in transparent mode in any commit) to correlate the results and see if MitMProxy could be used for our needs in a server. The problem is I'm not getting any QUIC packets in MitMProxy (transparent nor WireGuard mode), not only requesting from the local Caddy Server but also to Google, Facebook, etc; and I know I'm supposed to see the QUIC packets there (I see them in Wireshark).
I have the MitMProxy CA certificate installed in both my PC and the phone (Android) I was using to test the WireGuard mode (and making the requests to the web pages from there too by connecting to the PC by creating a Wi-Fi hotspot in the phone and connecting the PC there). Be advised, I'm just web browsing at this point through Chrome/[Chromite](https://github.com/uazo/cromite)/[Fennec](https://f-droid.org/packages/org.mozilla.fennec_fdroid/) (don't know if those last two could block anything, but I'm using Chrome as control too so...), not testing traffic from specific apps like Google Maps, Twitter or whatever - we know that for that use case we need to undo the certificate pinning but I'm not sure if it's needed for the web browsers too. We have tested using an iPhone too, but no luck either.

So at this point, do any of you have any suggestion as to at least try to make the capture of QUIC packets work?

Thanks in advance!

**Comment by mhils** [comment]
https://mitmproxy.org/posts/releases/mitmproxy10/ has some example invocations if that helps. :)

**Reply by josemariafr-upm** [reply]
Yes, I know about that, and the reverse mode works, but we are interested in the Transparent/WireGuard or (preferably) Regular mode as I commented in my [previous discussion](https://github.com/mitmproxy/mitmproxy/discussions/6397), and the problem is that it's not working for me the Transparent mode for QUIC with the commit that you said :_D

**Comment by josemariafr-upm** [comment]
Hi @mhils, any news on the transparent/regular mode with QUIC-HTTP/3 support? :D
